selfdrive/car/car_specific.py

```diff
@@ -216,7 +216,7 @@ def create_common_events(self, CS: structs.CarState, CS_prev: car.CarState, extr
       if not self.CP.pcmCruise and (b.type in enable_buttons and not b.pressed):
         events.add(EventName.buttonEnable)
       # Disable on rising and falling edge of cancel for both stock and OP long
-      if b.type == ButtonType.cancel:
+      if b.type == ButtonType.cancel and (allow_button_cancel or not self.CP.pcmCruise):
         events.add(EventName.buttonCancel)
```

Comment:
panda safety across the brands is quite split on checking the cancel button if stock long is controlling or not, and the way this previously worked was Hyundai-specific in not adding the button events, so this seems like a decent solution until we make all the panda safety match each other (don't check when stock long?)